src/nanoemoji/svg.py

```diff
@@ -322,45 +331,73 @@ def _add_glyph(svg: SVG, color_glyph: ColorGlyph, reuse_cache: ReuseCache):
-            transform = reuse.translate(-tx, -ty)
-            if transform != Affine2D.identity():
-                svg_use.attrib["transform"] = _svg_matrix(transform)
+    el_by_paint = {}
+    complete_paths = set()
+    for root in color_glyph.painted_layers:
+        for context in root.breadth_first():
+            if any(c == context.path[: len(c)] for c in complete_paths):
+                continue
+
+            parent_el = svg_g
+            if context.path:
+                parent_el = context.path[-1]
+
+            if isinstance(context.paint, PaintGlyph):
+                paint_glyph = cast(PaintGlyph, context.paint)
+                glyph_path = paint_glyph.glyph
+                reuse_key = _inter_glyph_reuse_key(
+                    reuse_cache.reuse_tolerance, view_box, context.paint
+                )
+                created_use = False
+                if reuse_key in reuse_cache.shapes:
+                    el = reuse_cache.shapes[reuse_key]
+                    source_path = el.attrib["d"]
+                    transform = affine_between(
+                        SVGPath(d=source_path),
+                        SVGPath(d=glyph_path),
+                        reuse_cache.reuse_tolerance,
+                    )
+                    if transform:
+                        _ensure_has_id(el)
+
+                        # we have an inter-glyph shape reuse: move the reused element to the outer
+                        # <defs> and replace its first occurrence with a <use>. Adobe Illustrator
+                        # doesn't support direct references between glyphs:
+                        # https://github.com/googlefonts/nanoemoji/issues/264#issuecomment-820518808
+                        if el not in svg_defs:
+                            svg_use = etree.Element("use", nsmap=svg.svg_root.nsmap)
+                            svg_use.attrib[
+                                _XLINK_HREF_ATTR_NAME
+                            ] = f'#{el.attrib["id"]}'
+                            el.addnext(svg_use)
+                            svg_defs.append(el)  # append moves
+
+                        svg_use = etree.SubElement(
+                            parent_el, "use", nsmap=svg.svg_root.nsmap
+                        )
+                        svg_use.attrib[_XLINK_HREF_ATTR_NAME] = f'#{el.attrib["id"]}'
+                        tx, ty = transform.gettranslate()
+                        if tx:
+                            svg_use.attrib["x"] = _ntos(tx)
+                        if ty:
+                            svg_use.attrib["y"] = _ntos(ty)
+                        transform = transform.translate(-tx, -ty)
+                        if transform != Affine2D.identity():
+                            svg_use.attrib["transform"] = _svg_matrix(transform)
+
+                        created_use = True
+
+                if not created_use:
+                    el = to_element(SVGPath(d=paint_glyph.glyph))
+                    _apply_paint(
```

Comment:
...should be checking transform here? - sanity check we have a test w/transforms